src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -28,14 +28,26 @@ var {
 var ReactFiber = require('ReactFiber');
 var ReactReifiedYield = require('ReactReifiedYield');
 
-function createSubsequentChild(parent : Fiber, previousSibling : Fiber, newChildren) : Fiber {
+function createSubsequentChild(parent : Fiber, nextReusable : ?Fiber, previousSibling : Fiber, newChildren) : Fiber {
   if (typeof newChildren !== 'object' || newChildren === null) {
     return previousSibling;
   }
 
   switch (newChildren.$$typeof) {
     case REACT_ELEMENT_TYPE: {
       const element = (newChildren : ReactElement<any>);
+      if (nextReusable &&
+          element.type === nextReusable.type &&
+          element.key === nextReusable.key) {
+        // TODO: This is not sufficient since previous siblings could be new.
+        // Will fix reconciliation properly later.
```

Comment:
In the unit tests this is currently true but not necessarily. `nextReusable` lines up with the index in the array but the key and type might not.